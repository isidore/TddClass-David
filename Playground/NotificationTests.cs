using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalUtilities.Persistence;
using eBabyServices;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Playground
{
    [TestClass]
    [UseReporter(typeof(DiffReporter))]

    public class NotificationTests
    {
        [TestMethod]
        public void TestNotifySellerNoBidders()
        {
            PostOffice.GetInstance().Clear();
            (Users users, User scott, User bob, Auction auction) =AuctionTests.CreateAuctionWorld();
            auction.StartAuction();
            auction.EndAuction();
            Approvals.Verify(PostOffice.GetInstance().FindEmail(scott.Email, auction.ItemDescription));
        }

        [TestMethod]
        public void TestNotificationEmailForSellerNoBids()
        {
            (Users users, User scott, User bob, Auction auction) =AuctionTests.CreateAuctionWorld();
            auction.StartAuction();
            
            Approvals.VerifyAll(auction.GetClosingEmailNotifications());
        }

        [TestMethod]
        public void TestNotifySellerThatTheItemSold()
        {
            //create auction world
            //start an auction
            //bob bids $5
            //verify closing notification emails
            (Users users, User scott, User bob, Auction auction) = AuctionTests.CreateAuctionWorld();
            auction.StartAuction();
            auction.Bid(bob, 5);
            Approvals.VerifyAll(auction.GetClosingEmailNotifications());
        }
    }

}

