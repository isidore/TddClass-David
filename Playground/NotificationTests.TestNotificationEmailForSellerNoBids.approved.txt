srkirkland@ucdavis.edu => Sorry, your auction for item description did not have any bidders.
